packages/manager/src/views/SPAPropertyDetailPage/StaticSPADeployment/CreateStaticApp/CreateStaticApp.tsx

```diff
@@ -0,0 +1,239 @@
+import {
+  Button,
+  Form,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  Split,
+  SplitItem,
+  TextInput,
+  FileUpload
+} from '@patternfly/react-core';
+import { AxiosError } from 'axios';
+import toast from 'react-hot-toast';
+import * as yup from 'yup';
+import { useGetWebPropertyGroupedByEnv, useCreateStaticApp } from '@app/services/persistent';
+
+interface CreateStaticAppProps {
+  onClose: () => void;
+  propertyIdentifier: string;
+}
+
+const schema = yup.object({
+  name: yup.string().required().label('Application Name'),
+  path: yup
+    .string()
+    .matches(/^[a-zA-Z0-9/-]+$/, 'Only letters, numbers, forward slash and dashes are allowed')
+    .required(),
+  env: yup.string().required().label('Environment'),
+  ref: yup.string().required().label('Reference'),
+  upload: yup
+    .mixed()
+    .required('A file is required')
+    .test('fileType', 'Invalid file type', (value) => {
+      const allowedExtensions = ['zip', 'tgz', 'gzip', 'gz', 'rar', 'tar'];
+      if (value && value.length > 0) {
+        const fileExtension = value[0]?.name.split('.').pop();
+        return allowedExtensions.includes(fileExtension);
+      }
+      return false;
+    })
+});
+
+type FormData = yup.InferType<typeof schema>;
+
+export const CreateStaticApp: React.FC<CreateStaticAppProps> = ({
+  onClose,
+  propertyIdentifier
+}) => {
+  const [fileValue, setFileValue] = useState<File | string>('');
+  const [filename, setFilename] = useState('');
+
+  const {
+    control,
+    handleSubmit,
+    setValue,
+    formState: { isSubmitting, errors },
+    trigger
+  } = useForm<FormData>({
+    defaultValues: { path: '/', upload: null },
+    mode: 'onBlur',
+    resolver: yupResolver(schema)
+  });
+
+  const webProperties = useGetWebPropertyGroupedByEnv(propertyIdentifier);
+  const webPropertiesKeys = Object.keys(webProperties.data || {});
+  const createStaticSpa = useCreateStaticApp();
+
+  const onSubmit = async (data: FormData) => {
+    try {
+      const formData = new FormData();
+      formData.append('propertyIdentifier', propertyIdentifier);
+      formData.append('name', data.name);
+      formData.append('path', data.path);
+      formData.append('env', data.env);
+      formData.append('ref', data.ref);
+
+      if (data.upload && data.upload[0]) {
+        formData.append('upload', data.upload[0], data.upload[0].name);
+      }
+
+      await createStaticSpa.mutateAsync(formData);
+      onClose();
+      toast.success('Deployed application successfully');
+    } catch (error) {
+      if (error instanceof AxiosError && error.response && error.response.status === 403) {
+        toast.error("You don't have access to perform this action");
+        onClose();
+      } else if (error instanceof AxiosError && error.response && error.response.status === 400) {
+        toast.error(
+          "Given Image URL doesn't exist on the source registry, please provide a valid image URL."
```

Comment:
`Given Image URL doesn't exist on the source registry, please provide a valid image URL.`

Please change it to `Please provide the valid input file`

Reply:
This is required at parent level at application listing page